examples/tutorial.rs

```diff
@@ -40,7 +40,7 @@ fn array_basics() {
     let y = Array::ones::<f32>(&[2, 2]).unwrap();
 
     // Pointwise add x and y:
-    let z = x.add(&y);
+    let z = x.as_ref().add(&y);
```

Comment:
is this change related?

Reply:
I thought the compiler would complain because `x` is not a ref, but `x.add(&y)` actually works fine. This change is reverted in [ac32a53](https://github.com/oxideai/mlx-rs/pull/71/commits/ac32a53e652fc819ef8a885f51e4a28c6231ba55)